src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -1851,4 +1875,20 @@ public override void Post(SendOrPostCallback d, object state)
             base.Post(d, state);
         }
     }
+
+
+    private class ExceptionThrowingFormatter : JsonMessageFormatter, IJsonRpcMessageFormatter
+    {
+        public bool ThrowException = false;
+
+        public new void Serialize(IBufferWriter<byte> bufferWriter, JsonRpcMessage message)
```

Comment:
I'm *really* surprised this works. the `new` modifier on this method should mean it's entirely independent of all base class methods, and thus I wouldn't expect it to ever be called by the library. How is this getting called?
In any case, I think overriding a method with `override` would be the more appropriate choice instead of `new`.

Reply:
The trick here is that I made this class also implement IJsonRpcMessageFormatter, is kind of a hacky way to get this called from the message handlers. Without that interface, this never gets called as you said. I tried overriding at first since it seemed like the best for me too, but I get an error saying that I can't override the serialize method because it is not marked as virtual, abstract or override. Should I mark the method in JsonMessageFormatter as override or virtual and use override here?

---
In reply to: [314126468](https://github.com/microsoft/vs-streamjsonrpc/pull/318#discussion_r314126468) [](ancestors = 314126468)

Reply:
Very clever. Your technique is fine. I just hadn't seen it used hardly ever so I didn't think of it to look at it.